panic when running golint

```
make golint   
ERRO [linters_context/goanalysis] nilness: panic during analysis: runtime error: index out of range [1] with length 1, goroutine 9700 [running]:
runtime/debug.Stack()
        runtime/debug/stack.go:24 +0x64
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyzeSafe.func1()
        github.com/golangci/golangci-lint@v1.55.2/pkg/golinters/goanalysis/runner_action.go:105 +0x4c
panic({0x101749700?, 0x14001a06048?})
        runtime/panic.go:914 +0x218
honnef.co/go/tools/analysis/facts/nilness.impl.func1({0x1017d60d8, 0x1400c93c690})
        honnef.co/go/tools@v0.4.6/analysis/facts/nilness/nilness.go:160 +0x900
honnef.co/go/tools/analysis/facts/nilness.impl.func1({0x1017d6008, 0x1400c942380})
        honnef.co/go/tools@v0.4.6/analysis/facts/nilness/nilness.go:147 +0x938
honnef.co/go/tools/analysis/facts/nilness.impl(0x1400be260d0, 0x1400bc0f7c0, 0x140015d5c88)
        honnef.co/go/tools@v0.4.6/analysis/facts/nilness/nilness.go:246 +0x2c8
honnef.co/go/tools/analysis/facts/nilness.run(0x1400be260d0)
        honnef.co/go/tools@v0.4.6/analysis/facts/nilness/nilness.go:66 +0xf4
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyze(0x14002f69f30)
        github.com/golangci/golangci-lint@v1.55.2/pkg/golinters/goanalysis/runner_action.go:195 +0x900
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyzeSafe.func2()
        github.com/golangci/golangci-lint@v1.55.2/pkg/golinters/goanalysis/runner_action.go:113 +0x20
github.com/golangci/golangci-lint/pkg/timeutils.(*Stopwatch).TrackStage(0x14001960000, {0x10127daa1, 0x7}, 0x140037e2f30)
        github.com/golangci/golangci-lint@v1.55.2/pkg/timeutils/stopwatch.go:111 +0x44
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyzeSafe(0x100c40350?)
        github.com/golangci/golangci-lint@v1.55.2/pkg/golinters/goanalysis/runner_action.go:112 +0x78
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).analyze.func2(0x14002f69f30)
        github.com/golangci/golangci-lint@v1.55.2/pkg/golinters/goanalysis/runner_loadingpackage.go:80 +0xb0
created by github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).analyze in goroutine 1177
        github.com/golangci/golangci-lint@v1.55.2/pkg/golinters/goanalysis/runner_loadingpackage.go:75 +0x184 
ERRO [runner] Panic: nilness: package "transport" (isInitialPkg: false, needAnalyzeSource: true): runtime error: index out of range [1] with length 1: goroutine 9700 [running]:
runtime/debug.Stack()
        runtime/debug/stack.go:24 +0x64
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyzeSafe.func1()
        github.com/golangci/golangci-lint@v1.55.2/pkg/golinters/goanalysis/runner_action.go:109 +0x224
panic({0x101749700?, 0x14001a06048?})
        runtime/panic.go:914 +0x218
honnef.co/go/tools/analysis/facts/nilness.impl.func1({0x1017d60d8, 0x1400c93c690})
        honnef.co/go/tools@v0.4.6/analysis/facts/nilness/nilness.go:160 +0x900
honnef.co/go/tools/analysis/facts/nilness.impl.func1({0x1017d6008, 0x1400c942380})
        honnef.co/go/tools@v0.4.6/analysis/facts/nilness/nilness.go:147 +0x938
honnef.co/go/tools/analysis/facts/nilness.impl(0x1400be260d0, 0x1400bc0f7c0, 0x140015d5c88)
        honnef.co/go/tools@v0.4.6/analysis/facts/nilness/nilness.go:246 +0x2c8
honnef.co/go/tools/analysis/facts/nilness.run(0x1400be260d0)
        honnef.co/go/tools@v0.4.6/analysis/facts/nilness/nilness.go:66 +0xf4
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyze(0x14002f69f30)
        github.com/golangci/golangci-lint@v1.55.2/pkg/golinters/goanalysis/runner_action.go:195 +0x900
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyzeSafe.func2()
        github.com/golangci/golangci-lint@v1.55.2/pkg/golinters/goanalysis/runner_action.go:113 +0x20
github.com/golangci/golangci-lint/pkg/timeutils.(*Stopwatch).TrackStage(0x14001960000, {0x10127daa1, 0x7}, 0x140037e2f30)
        github.com/golangci/golangci-lint@v1.55.2/pkg/timeutils/stopwatch.go:111 +0x44
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyzeSafe(0x100c40350?)
        github.com/golangci/golangci-lint@v1.55.2/pkg/golinters/goanalysis/runner_action.go:112 +0x78
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).analyze.func2(0x14002f69f30)
        github.com/golangci/golangci-lint@v1.55.2/pkg/golinters/goanalysis/runner_loadingpackage.go:80 +0xb0
created by github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).analyze in goroutine 1177
        github.com/golangci/golangci-lint@v1.55.2/pkg/golinters/goanalysis/runner_loadingpackage.go:75 +0x184 
WARN [runner] Can't run linter goanalysis_metalinter: goanalysis_metalinter: nilness: package "transport" (isInitialPkg: false, needAnalyzeSource: true): runtime error: index out of range [1] with length 1 
ERRO Running error: 1 error occurred:
        * can't run linter goanalysis_metalinter: goanalysis_metalinter: nilness: package "transport" (isInitialPkg: false, needAnalyzeSource: true): runtime error: index out of range [1] with length 1
 
make[2]: *** [lint] Error 3
make[1]: *** [cmd/otelcorecol] Error 2
make: *** [golint] Error 2
```

--- a/internal/tools/go.mod
+++ b/internal/tools/go.mod
@@ -5,7 +5,7 @@ go 1.20
 require (
 	github.com/a8m/envsubst v1.4.2
 	github.com/client9/misspell v0.3.4
-	github.com/golangci/golangci-lint v1.55.2
+	github.com/golangci/golangci-lint v1.55.1
 	github.com/google/addlicense v1.1.1
 	github.com/jcchavezs/porto v0.6.0
 	github.com/mikefarah/yq/v4 v4.40.1

